singer_sdk/connectors/sql.py

```diff
@@ -779,14 +779,22 @@ def merge_sql_types(
                     (sqlalchemy.types.String, sqlalchemy.types.Unicode),
                 ):
                     # If length None or 0 then is varchar max ?
-                    if (opt_len is None) or (opt_len == 0):
+                    if (
+                        (opt_len is None)
+                        or (opt_len == 0)
+                        or (opt_len >= (current_type.length or (opt_len + 1)))
```

Comment:
Is the last condition there in case `current_type.length` is None? In that case it's a nit bit hard to read it like that. How about

```suggestion
                        or (current_type.length and (opt_len >= current_type.length))
```

Reply:
Not exactly.  The issue is when `current_type.length is None` that means it defaults to a size of `MAX`.  Since `MAX` will be larger than any size found in `opt_len` I used `or` to fill in `None` with `(opt_len +1)` so it will choose to keep the columns current `sql_type`.  Before this PR it would choose the opt column with a `opt_len` of `1` over the columns current size of `None or MAX` .  I know clear as mud.

Reply:
@BuzzCutNorman I see, thanks for your response. Would my suggestion have the undesired effect then?

Reply:
@edgarrmondragon I'm a dork.  Yes, your code would have the desired effect.